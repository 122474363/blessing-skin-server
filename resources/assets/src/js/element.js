import Vue from 'vue'
import Message from 'element-ui/lib/message'
import MessageBox from 'element-ui/lib/message-box'

Vue.prototype.$message = Message
Vue.prototype.$msgbox = MessageBox
Vue.prototype.$alert = MessageBox.alert
Vue.prototype.$confirm = MessageBox.confirm
Vue.prototype.$prompt = MessageBox.prompt

window.MessageBox = MessageBox

export {
  Message,
  MessageBox,
}
