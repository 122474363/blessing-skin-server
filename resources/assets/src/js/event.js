/** @type {{ [name: string]: Function[] }} */
const bus = Object.create(null);

/**
 * @param {string} eventName
 * @param {Function} listener
 */
export function on(eventName, listener) {
    (bus[eventName] || (bus[eventName] = [])).push(listener);
}

/**
 * @param {string} eventName
 * @param {any} payload
 */
export function emit(eventName, payload) {
    bus[eventName] && bus[eventName].forEach(listener => listener(payload));
}

Object.defineProperty(window, 'bsEmitter', {
    get: () => Object.freeze({ on, emit })
});
