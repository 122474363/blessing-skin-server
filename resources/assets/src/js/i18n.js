import Vue from 'vue';

/**
 * Translate according to given key.
 *
 * @param  {string} key
 * @param  {object} parameters
 * @return {string}
 */
export function trans(key, parameters = Object.create(null)) {
    const segments = key.split('.');
    let temp = blessing.i18n || Object.create(null);

    for (const segment of segments) {
        if (!temp[segment]) {
            return key;
        } else {
            temp = temp[segment];
        }
    }

    Object.keys(parameters)
        .forEach(slot => temp = temp.replace(`:${slot}`, parameters[slot]));

    return temp;
}

Vue.use(_Vue => {
    _Vue.prototype.$t = trans;
    _Vue.directive('t', (el, { value }) => {
        if (typeof value === 'string') {
            el.textContent = trans(value);
        } else if (typeof value === 'object') {
            el.textContent = trans(value.path, value.args);
        } else {
            /* istanbul ignore next */
            if (process.env.NODE_ENV !== 'production') {
                console.warn('[i18n] Invalid arguments in `v-t` directive.');
            }
        }
    });
});

window.trans = trans;
