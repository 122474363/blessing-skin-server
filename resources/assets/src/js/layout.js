import './jquery' // jQuery first
import 'bootstrap'
import 'admin-lte'
import 'icheck'
import Vue from 'vue'

Vue.mixin({
  mounted() {
    $(this.$el).iCheck({
      radioClass: 'iradio_square-blue',
      checkboxClass: 'icheckbox_square-blue',
    })
      .on('ifChecked ifUnchecked', function onChange() {
        // eslint-disable-next-line no-invalid-this
        $(this)[0].dispatchEvent(new Event('change'))
      })
    $('[data-toggle="tooltip"]').tooltip()
  },
})

$(document).ready(() => {
  $('input').iCheck({
    radioClass: 'iradio_square-blue',
    checkboxClass: 'icheckbox_square-blue',
  })

  $('[data-toggle="tooltip"]').tooltip()
})
;(() => {
  const list = [
    {
      path: 'admin',
      styl: () => import('../stylus/admin.styl'),
    },
    {
      path: 'auth',
      styl: () => import('../stylus/auth.styl'),
    },
  ]
  const style = list.find(({ path }) => blessing.route.startsWith(path))
  if (style) {
    style.styl()
  }
})()
