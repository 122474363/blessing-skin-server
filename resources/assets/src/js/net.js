import Vue from 'vue';
import { emit } from './event';
import { queryStringify } from './utils';
import { showAjaxError } from './notify';

const empty = Object.create(null);
/** @type Request */
export const init = {
    credentials: 'same-origin',
    headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
    }
};

function retrieveToken() {
    const csrfField = document.querySelector('meta[name="csrf-token"]');
    return csrfField && csrfField.content;
}

/**
 * @param {Request} request
 */
export async function walkFetch(request) {
    request.headers.set('X-CSRF-TOKEN', retrieveToken());

    try {
        const response = await fetch(request);
        if (response.ok) {
            return response.json();
        } else {
            showAjaxError(await response.text());
        }
    } catch (error) {
        showAjaxError(error);
    }
}

export async function get(url, params = empty) {
    emit('beforeFetch', {
        method: 'GET',
        url,
        data: params
    });

    const qs = queryStringify(params);

    return walkFetch(new Request(`${blessing.base_url}${url}${qs && '?' + qs}`, init));
}

export async function post(url, data = empty) {
    emit('beforeFetch', {
        method: 'POST',
        url,
        data
    });

    return walkFetch(new Request(`${blessing.base_url}${url}`, {
        body: JSON.stringify(data),
        method: 'POST',
        ...init
    }));
}

Vue.use(_Vue => {
    _Vue.prototype.$http = {
        get,
        post,
    };
});

window.bsAjax = { get, post };
