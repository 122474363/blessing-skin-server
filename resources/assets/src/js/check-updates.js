import { init } from './net'

export async function checkForUpdates() {
  const response = await fetch(`${blessing.base_url}/admin/update/check`, init)

  if (response.ok) {
    const data = await response.json()
    if (data.available) {
      document.querySelector(`[href="${blessing.base_url}/admin/update"]`)
        .innerHTML += `
        <span class="pull-right-container">
          <span class="label label-primary pull-right">v${data.latest}</span>
        </span>`
    }
  }
}

export async function checkForPluginUpdates() {
  const response = await fetch(`${blessing.base_url}/admin/plugins/market/check`, init)

  if (response.ok) {
    const data = await response.json()
    if (data.available) {
      document.querySelector(`[href="${blessing.base_url}/admin/plugins/market"]`)
        .innerHTML += `
        <span class="pull-right-container">
          <span class="label label-success pull-right">${data.plugins.length}</span>
        </span>`
    }
  }
}

window.checkForUpdates = checkForUpdates
window.checkForPluginUpdates = checkForPluginUpdates
