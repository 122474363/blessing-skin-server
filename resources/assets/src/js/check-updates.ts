import { init } from './net'

export async function checkForUpdates(): Promise<void> {
  const response = await fetch(`${blessing.base_url}/admin/update/check`, init)

  if (response.ok) {
    const data = await response.json()
    const el = document.querySelector(`[href="${blessing.base_url}/admin/update"]`)
    if (data.available && el) {
      el.innerHTML += `
        <span class="pull-right-container">
          <span class="label label-primary pull-right">v${data.latest}</span>
        </span>`
    }
  }
}

export async function checkForPluginUpdates(): Promise<void> {
  const response = await fetch(`${blessing.base_url}/admin/plugins/market/check`, init)

  if (response.ok) {
    const data = await response.json()
    const el = document.querySelector(`[href="${blessing.base_url}/admin/plugins/market"]`)
    if (data.available && el) {
      el.innerHTML += `
        <span class="pull-right-container">
          <span class="label label-success pull-right">${data.plugins.length}</span>
        </span>`
    }
  }
}

Object.assign(window, {
  checkForUpdates,
  checkForPluginUpdates,
})
