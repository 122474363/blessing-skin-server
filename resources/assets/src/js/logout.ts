import { post } from './net'
import { swal } from './notify'
import { trans } from './i18n'

export async function logout() {
  const { dismiss } = await swal({
    text: trans('general.confirmLogout'),
    type: 'warning',
    showCancelButton: true,
    confirmButtonText: trans('general.confirm'),
    cancelButtonText: trans('general.cancel'),
  })
  if (dismiss) {
    return
  }

  const { msg } = await post('/auth/logout')
  setTimeout(() => (window.location.href = blessing.base_url), 1000)
  swal({ type: 'success', text: msg })
}

const button = document.querySelector('#logout-button')
/* istanbul ignore next, not all pages contains this button. */
if (button) {
  button.addEventListener('click', logout)
}
