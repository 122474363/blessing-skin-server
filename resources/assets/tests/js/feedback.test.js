import { sendFeedback } from '@/js/feedback';

test('send feedback', async () => {
    window.fetch = jest.fn()
        .mockResolvedValueOnce({ ok: false })
        .mockResolvedValueOnce({
            ok: true,
            json: () => Promise.resolve({ errno: 1 })
        })
        .mockResolvedValue({
            ok: true,
            json: () => Promise.resolve({ errno: 0 })
        });

    await sendFeedback();
    expect(document.cookie).toBe('');
    expect(fetch.mock.calls[0]).toMatchSnapshot();

    await sendFeedback();
    expect(document.cookie).toBe('');

    await sendFeedback();
    expect(document.cookie).toStartWith('feedback_sent=');

    window.fetch.mockClear();
    await sendFeedback();
    expect(window.fetch).not.toBeCalled();
});
