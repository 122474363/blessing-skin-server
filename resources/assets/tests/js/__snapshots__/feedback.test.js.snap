// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`send feedback 1`] = `
Array [
  "https://work.prinzeugen.net/statistics/feedback",
  Object {
    "body": "site_name=Blessing%20Skin&site_url=&version=4.0.0",
    "headers": Object {
      "Accept": "application/json",
      "Content-Type": "application/x-www-form-urlencoded",
    },
    "method": "POST",
    "mode": "cors",
  },
]
`;
